### ISSUE TITLE
Timestamp 10years problem

### ISSUE THREAD

**Issue by robots** [pigrew/trueposctrl#6]
Hi,

Your 10 year problem is that GPS time is calculated from 5. january 1980.

So this is not correct
`
        // GPS UNIX-timestamp (secs since 1970), but my board is off by 10 years (reporting 2007 while it is 2017)
`
as it is not GPS UNIX timestamp, but GPS GPS timestamp :-)

`        // GPS seconds since 5. january 1980`

**Comment by pigrew** [comment]
Ah, yes, the comment should be corrected. Thanks for reviewing it.

I don't have my board in front of me to check exactly what it gives. It might take a week or two for me to review it myself.

When I was writing the code, I had assumed that it was a week-overflow problem causing the offset, but it seems that's not the case.

Based on [this leapsecond site](http://leapsecond.com/java/gpsclock.htm#:~:text=GPS%20time%20was%20zero%20at,Loran%2DC%20chain%20transmitter%20sites.), GPS timestamps should start 6 Jan 1980 and shouldn't have been adjusted with leap seconds.

Were you able to verify that `$CLOCK` actually is GPS time (no leap seconds, starts 6 Jan 1980?)? How does the following statement look to you?

`// GPS time (seconds since 6-Jan-1980 0h00 UTC, without leap second corrections). This is corrected for week overflow as of board manufacture. It likely will be in error due to week number overflows, about 20 years after manufacture.`

**Comment by robots** [comment]
yes, i have done similar project as you , but i forgot to push it into github :-)

i use function like this to get unix timestamp:
uint32_t gps_getunixtime(void)
{
        uint32_t out = gps_clock.time;
        out += 315964800;
        out -= gps_clock.leap;
        return out;
}
So far i have not had GPS rollover problem. Date is still correct.

My project is here: https://github.com/robots/trueposition_gpsdo
